molgenis-api-permissions/src/main/java/org/molgenis/api/permissions/PermissionsController.java

```diff
@@ -0,0 +1,473 @@
+import org.molgenis.api.permissions.model.request.PermissionRequest;
+import org.molgenis.api.permissions.model.request.SetObjectPermissionRequest;
+import org.molgenis.api.permissions.model.request.SetTypePermissionsRequest;
+import org.molgenis.api.permissions.model.response.AllPermissionsResponse;
+import org.molgenis.api.permissions.model.response.LabelledPermissionResponse;
+import org.molgenis.api.permissions.model.response.ObjectPermissionResponse;
+import org.molgenis.api.permissions.model.response.ObjectResponse;
+import org.molgenis.api.permissions.model.response.PermissionResponse;
+import org.molgenis.api.permissions.model.response.TypePermissionsResponse;
+import org.molgenis.api.permissions.model.response.TypeResponse;
+import org.molgenis.api.permissions.rsql.PermissionRsqlVisitor;
+import org.molgenis.api.permissions.rsql.PermissionsQuery;
+import org.molgenis.data.security.permission.EntityHelper;
+import org.molgenis.data.security.permission.PermissionService;
+import org.molgenis.data.security.permission.PermissionSetUtils;
+import org.molgenis.data.security.permission.UserRoleTools;
+import org.molgenis.data.security.permission.model.LabelledPermission;
+import org.molgenis.data.security.permission.model.LabelledType;
+import org.molgenis.data.security.permission.model.Permission;
+import org.molgenis.security.acl.ObjectIdentityService;
+import org.molgenis.security.core.PermissionSet;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.acls.model.Sid;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Api("Permissions API")
+@RestController
+@RequestMapping(BASE_URI)
+@Transactional
+public class PermissionsController extends ApiController {
+
+  private static final String PERMISSION_API_IDENTIFIER = "permissions";
+  public static final String BASE_URI = "/api/" + PERMISSION_API_IDENTIFIER;
+  static final Integer DEFAULT_PAGE = 1;
+  static final Integer DEFAULT_PAGESIZE = 100;
+
+  private static final String TYPE = "type";
+  public static final String TYPES = TYPE + "s";
+  private static final String TYPE_ID = TYPE + "Id";
+
+  private static final String OBJECT = "object";
+  public static final String OBJECTS = OBJECT + "s";
+  private static final String OBJECT_ID = OBJECT + "Id";
+
+  private final PermissionService permissionService;
+  private final RSQLParser rsqlParser;
```

Comment:
Can the ```cz.jirutka.rsql``` dependency be removed by using QueryRsql (see RestControllerV2)?

Reply:
Don't think so, all our RSQL code is based on "<Entity> queries"